# First part by Witold @wdyrka in rstudio ghp#1063
diff --git a/src/cpp/core/http/SocketProxy.cpp b/src/cpp/core/http/SocketProxy.cpp
index 6d352b6e3b..0c78401c68 100644
--- a/src/cpp/core/http/SocketProxy.cpp
+++ b/src/cpp/core/http/SocketProxy.cpp
@@ -148,8 +148,12 @@ namespace {
 #ifndef _WIN32
 bool isSslShutdownError(const core::Error& error)
 {
+   #if defined SSL_R_SHORT_READ
    return error.code().category() == boost::asio::error::get_ssl_category() &&
           error.code().value() == ERR_PACK(ERR_LIB_SSL, 0, SSL_R_SHORT_READ);
+   #else
+   return error.code() == boost::asio::ssl::error::stream_truncated;  
+   #endif
 }
 #else
 bool isSslShutdownError(const core::Error& error)
