diff --git a/src/node/desktop/CMakeLists.txt b/src/node/desktop/CMakeLists.txt
index cdc23398fb..248550799d 100644
--- a/src/node/desktop/CMakeLists.txt
+++ b/src/node/desktop/CMakeLists.txt
@@ -201,7 +201,7 @@ elseif(LINUX)
    endif()
 
    install(
-      DIRECTORY "${ELECTRON_BUILD_DIR}/out/rstudio-linux-${ELECTRON_ARCH}/"
+      DIRECTORY "${ELECTRON_BUILD_DIR}/out/RStudio-linux-${ELECTRON_ARCH}/"
       DESTINATION "${RSTUDIO_INSTALL_ELECTRON}"
       USE_SOURCE_PERMISSIONS
    )
