diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 9fff318025..b8c78a548a 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -273,13 +273,13 @@ endif()
 
 # allow opt-in to using Boost.Signals2
 # TODO: remove this in RStudio v1.3 and port to signals2
-if(NOT RSTUDIO_BOOST_SIGNALS_VERSION)
-   if (Boost_VERSION LESS 106900)
-      set(RSTUDIO_BOOST_SIGNALS_VERSION 1)
-   else()
-      set(RSTUDIO_BOOST_SIGNALS_VERSION 2)
-   endif()
-endif()
+# if(NOT RSTUDIO_BOOST_SIGNALS_VERSION)
+#    if (Boost_VERSION LESS 106900)
+#       set(RSTUDIO_BOOST_SIGNALS_VERSION 1)
+#    else()
+       set(RSTUDIO_BOOST_SIGNALS_VERSION 2)
+#    endif()
+# endif()
 
 message(STATUS "Using Boost.Signals version ${RSTUDIO_BOOST_SIGNALS_VERSION}")
 add_definitions(-DRSTUDIO_BOOST_SIGNALS_VERSION=${RSTUDIO_BOOST_SIGNALS_VERSION})
@@ -359,8 +359,8 @@ endif()
 
 # SOCI
 if(UNIX)
-   set(RSTUDIO_TOOLS_SOCI "/opt/rstudio-tools/soci")
-   set(SOCI_INCLUDE_BUILD_DIR "${RSTUDIO_TOOLS_SOCI}/build/include")
+   set(RSTUDIO_TOOLS_SOCI "/usr/lib")
+   set(SOCI_INCLUDE_BUILD_DIR "/usr/include/soci")
 else()
    set(RSTUDIO_TOOLS_SOCI "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/soci")
    if(RSTUDIO_SESSION_WIN32)
@@ -393,7 +393,7 @@ endif()
 
 # find SOCI libraries
 if(UNIX)
-   set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}/build/lib")
+   set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}")
    set(CMAKE_INSTALL_RPATH "${SOCI_LIBRARY_DIR}")
    if (NOT APPLE)
       set(LIB_SUFFIX ".so")
@@ -404,8 +404,8 @@ if(UNIX)
       set(SOCI_LIB_SYMLINK ".4.0${LIB_SUFFIX}")
       set(SOCI_LIB_SUFFIX ".4.0.0${LIB_SUFFIX}")
    endif()
-   file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/*${LIB_SUFFIX}")
-   file(GLOB_RECURSE SOCI_SYMLINK_LIBRARIES "${SOCI_LIBRARY_DIR}/*${SOCI_LIB_SYMLINK}")
+   file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/libsoci_*${LIB_SUFFIX}")
+   file(GLOB_RECURSE SOCI_SYMLINK_LIBRARIES "${SOCI_LIBRARY_DIR}/libsoci_*${SOCI_LIB_SYMLINK}")
 else()
    set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}/build/${SOCI_ARCH}/lib")
    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
