diff --git a/src/cpp/desktop/DesktopGwtCallback.cpp b/src/cpp/desktop/DesktopGwtCallback.cpp
index 6132f238e5..31c445294f 100644
--- a/src/cpp/desktop/DesktopGwtCallback.cpp
+++ b/src/cpp/desktop/DesktopGwtCallback.cpp
@@ -79,7 +79,7 @@ GwtCallback::GwtCallback(MainWindow* pMainWindow, GwtWindow* pOwner)
    // via environment variable. clipboard monitoring enables us to support middle-click paste on
    // Linux, but it causes problems on some systems.
    if (desktop::options().clipboardMonitoring() &&
-       core::system::getenv("RSTUDIO_NO_CLIPBOARD_MONITORING").empty()
+       core::system::getenv("RSTUDIO_NO_CLIPBOARD_MONITORING").empty())
    {
       QClipboard* clipboard = QApplication::clipboard();
       if (clipboard->supportsSelection())
