diff --git a/dependencies/common/install-copilot-language-server b/dependencies/common/install-copilot-language-server
index a2f70ac245..dec188c5d2 100755
--- a/dependencies/common/install-copilot-language-server
+++ b/dependencies/common/install-copilot-language-server
@@ -25,7 +25,7 @@ COPILOT_VERSION="1.296.0"
 
 # determine platform and architecture
 PLATFORM="$(uname)-$(getconf LONG_BIT)"
-if [ "$(arch)" == "arm64" ] || [ "$(arch)" == "aarch64" ]; then
+if [ "$(uname -m)" == "arm64" ] || [ "$(uname -m)" == "aarch64" ]; then
   ARCH="arm64"
 else
   ARCH="x64"
