--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -108,26 +108,29 @@
    -DQT_NO_SIGNALS_SLOTS_KEYWORDS)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-find_package(Qt5Core REQUIRED)
-find_package(Qt5Widgets REQUIRED)
-find_package(Qt5Gui REQUIRED)
-find_package(Qt5Network REQUIRED)
-find_package(Qt5WebChannel REQUIRED)
+find_package(Qt5 REQUIRED COMPONENTS
+             Core
+             Widgets
+             Gui
+             Network
+             WebChannel
+             PrintSupport
+             Quick
+             QuickWidgets
+             Positioning
+             Sensors
+             Svg
+             Xml
+             XmlPatterns
+             Qml
+             Sql
+             OpenGL
+)
 find_package(Qt5WebEngine REQUIRED)
 find_package(Qt5WebEngineWidgets REQUIRED)
-find_package(Qt5PrintSupport REQUIRED)
-find_package(Qt5Quick REQUIRED)
-find_package(Qt5QuickWidgets REQUIRED)
-find_package(Qt5Positioning REQUIRED)
-find_package(Qt5Sensors REQUIRED)
-find_package(Qt5Svg REQUIRED)
-find_package(Qt5Xml REQUIRED)
-find_package(Qt5XmlPatterns REQUIRED)
-find_package(Qt5Qml REQUIRED)
-find_package(Qt5Sql REQUIRED)
-find_package(Qt5OpenGL REQUIRED)
 if(UNIX AND NOT APPLE)
-   find_package(Qt5DBus REQUIRED)
+   find_package(Qt5 REQUIRED
+                DBus)
 endif()
 
 get_filename_component(QT_BINARY_PATH  "${QT_BIN_DIR}"            ABSOLUTE CACHE)
